### PR TITLE
Implement Chalk::Config#reload.

### PR DESCRIPTION
**Pull request by nelhage**
Add the ability to reload a specified configuration file.

This is useful for applications that need to pick up configuration from
an external source (e.g. puppet) and want to be able to gracefully
reload.

Add an interface to reload a single file, instead of everything, since
supporting targetted reload seems much saner than potentially having all
of your runtime configuration change out from under you on disk, but I'm
open to discussion on this point.

### CODE REVIEW

**Comment by nelhage** [comment]
r? @franklinhu 
cc @mcfunley since this may be relevant to your feature-flag work?

**Comment by mcfunley** [comment]
Great, I should be able to use this. Thanks, lgtm

**Comment by nelhage** [comment]
merging on @mcfunley's lgtm

File: lib/chalk-config.rb [hunk 1]
```diff
@@ -193,13 +206,33 @@ def register(filepath, options)
...
+    rescue Errno::ENOENT
+      raise unless options[:optional]
+    end
+
+    validate_config(registration.merge(config: config))
+
+    registration[:config] = config
+
+    allow_configatron_changes do
+      reapply_config
```
Comment (franklinhu) [inline lib/chalk-config.rb]: It looks like `reapply_config` calls `allow_configatron_changes` internally?

**Comment by franklinhu** [comment]
One nit, but otherwise lgtm

**Pull request merged**